.circleci/config.yml

```diff
@@ -0,0 +1,148 @@
+# CircleCI 2.0 configuration file for running Python on macOS.
+#
+# Email notifications are configured only from web.
+# See "Email Preferences by Organization" in
+# https://circleci.com/account/notifications
+#
+# Based on these v1.0 steps:
+# https://pewpewthespells.com/blog/building_python_on_circleci.html
+#
+# This uses the v2 workflows and a separate commit status is sent for each
+# job.
+#
+# CircleCI 2.0 automatically sets a $BASH_ENV variable to a
+# random name in /tmp, and will source this file for each step.
+#
+version: 2
+
+jobs:
+  static_checkers:
+    working_directory: ~/static-checkers
+    docker:
+      - image: circleci/python:2.7
+    steps:
+      # Get the source.
+      - checkout
+
+      - run:
+          name: Prepare the environment.
+          command: |
+            python --version
+            pip install -q --user --ignore-installed --upgrade virtualenv
+            pip install -q tox --user
+            echo 'export PATH=~/.local/bin:$PATH' >> $BASH_ENV
+
+      # Run each checker in a separate step.
+      - run:
+          name: Check newsfragment.
+          command: |
+            tox -r -e newsfragment
+      - run:
+          name: Check manifest-checker.
+          command: |
+            tox -r -e manifest-checker
+      - run:
+          name: Check pyflakes.
+          command: |
+            tox -r -e pyflakes
+
+
+  #
+  # Documentation tests are slow so we execute them in a separate job.
+  #
+  documentation:
+    working_directory: ~/documentation
+    docker:
+      - image: circleci/python:2.7
+    steps:
+      # Get the source.
+      - checkout
+
+      - run:
+          name: Prepare the environment.
+          command: |
+            python --version
+            pip install -q --user --ignore-installed --upgrade virtualenv
+            pip install -q tox --user
+            echo 'export PATH=~/.local/bin:$PATH' >> $BASH_ENV
+
+      - run:
+          name: Check the narrative documentation.
+          command: |
+            tox -r -e narrativedocs
+
+      - run:
+          name: Check the API documentation.
+          command: |
+            tox -r -e apidocs
+
+
+  #
+  # We run pyflakes with Python 3 .
+  #
+  pyflakes3:
```

Comment:
Shouldn't this also be a static check, like `pyflakes` under Python 2?

Reply:
I don't understand the question. This is a separate job as it needs a separate docker image.

Reply:
Yes... This could have been a pre-requisite for the macOS build... but I thought that the python2 pyflakes should be enough ... so pyflakes3 and documentation builds are executed after that, to not delay the macOS any longer.

This is a compromise

Reply:
In the same times, this needs python3, so it would run as a separate job anyway.

In theory, we can create a python2+python3 docker image and run the static checkers with that image...but at this stage, I prefer to use the Circle-CI Docker images.